handlers/deploy.go

```diff
@@ -187,11 +187,15 @@ func buildEnv(envProcess string, envVars map[string]string) []string {
 	return env
 }
 
-// BuildEncodedAuthConfig for private registry
+// BuildEncodedAuthConfig parses the image name for a repository, user name, and image name
+// If a repository is not included (ie: username/function-name), 'docker.io/' will be prepended
 func BuildEncodedAuthConfig(basicAuthB64 string, dockerImage string) (string, error) {
-	// extract registry server address
+	// use docker.io if no repository was included
+	if len(strings.Split(dockerImage, "/")) < 3 {
```

Comment:
I previously gave the feedback that we shouldn't use a magic number `3` - it may have been on the CLI repo.

What does 3 mean and how can we make sure it's clear going forward?

Reply:
My next PR in this repo will update that to be more clear.

|1|2|3|
|---|---|---|
|repository|username|image|

So, if there are less than 3, there's no repository included. During testing of this PR, I did also verify that this will still work as expected with _only_ an image name `function-1` for example

Reply:
Lets not over-think this.  Magic numbers should, as a minimum, be extracted into constants and given a meaningful name to assist readability.  `const fullImagePathLength int = 3` turns the existing code into `if len(strings.Split(dockerImage, "/")) < fullImagePathLength {` which even at a minimal level is more readable.

Taking it further we could wrap the left hand side into a fn called imagePathLength() we get:
```
if imagePathLength(dockerImage) < fullImagePathLength {
```

I lack imagination when naming things but I hope this helps demonstrate the point.

Reply:
`fullImagePath` means nothing to a reader - the suggestion was to express this differently - via two named variables for 1 and 2 showing that 1 was username (namespace) and 2 was including a domain name.

Reply:
Apologies, didn't see the suggestion only the question.